students/stellie/lesson02/assignment/charges_calc.py

```diff
@@ -0,0 +1,141 @@
+# Stella Kim
+# Assignment 2: Logging & Debugging
+
+"""Returns total price paid for individual rentals"""
+
+import argparse
+import json
+import datetime
+import math
+import logging
+import sys
+
+
+def logging_handler(level):
+    """
+    Setup logging with the following debug level requirements:
+    # 0: No debug messages or log file.
+    # 1: Only error messages.
+    # 2: Error messages and warnings.
+    # 3: Error messages, warnings and debug messages.
+
+    Debug level should contain general comments, warnings display missing
+    elements and errors are shown for inconsistencies in source data.
+    """
+
+    # Format logs
+    log_format = '%(asctime)s %(filename)s:%(lineno)-3d ' +\
+                 '%(levelname)s %(message)s'
+    log_file = datetime.datetime.now().strftime('%Y-%m-%d')+'.charges_calc.log'
+
+    # Setup logging format
+    formatter = logging.Formatter(log_format)
+
+    # Setup file handler
+    file_handler = logging.FileHandler(log_file)
+    file_handler.setLevel(logging.WARNING)
+    file_handler.setFormatter(formatter)
+
+    # Setup console handler
+    console_handler = logging.StreamHandler()
+    console_handler.setLevel(logging.DEBUG)
+    console_handler.setFormatter(formatter)
+
+    logger = logging.getLogger()
+    log_levels = {0: logging.NOTSET, 1: logging.ERROR, 2: logging.WARNING,
+                  3: logging.DEBUG}
+    try:
+        logger.setLevel(log_levels[level])
+    except KeyError:
+        print('Debugging level is invalid.  Please enter a level between 0-3.')
+
```

Comment:
Hi Stella,

Looks good. On this exception, force your code to exit, otherwise it will issue a warning on screen, which will not really be visible as it will be followed by log output. One option is to use _sys.exit()_ for it. You can submit an updated pull request with this change to get the full grade.

-Luis